inference-engine/src/mkldnn_plugin/nodes/ctc_loss.cpp

```diff
@@ -0,0 +1,348 @@
+                        [&](size_t targetIdx, size_t start, size_t end, float prevLogProb) {
+                    if (end > actualLogitLen) {
+                        if (sumPerThread[ithr] == -float_inf) {
+                            sumPerThread[ithr] = prevLogProb;
+                        } else if (prevLogProb != -float_inf) {
+                            if (sumPerThread[ithr] > prevLogProb)
+                                sumPerThread[ithr] = sumPerThread[ithr] + std::log1pf(std::exp(prevLogProb - sumPerThread[ithr]));
+                            else
+                                sumPerThread[ithr] = prevLogProb + std::log1pf(std::exp(sumPerThread[ithr] - prevLogProb));
+                        }
+                        return;
+                    }
+
+                    size_t nextIdx = targetIdx + 1;
+                    int64_t st64 = start;
+                    float newLogProb = prevLogProb;
+                    if (!_ctcMergeRepeated) {
+                        for (size_t pos = start; pos < end; pos++) {
+                            newLogProb = prevLogProb;
+                            for (size_t bl = start; bl < pos; bl++) {
+                                newLogProb += logProbabilities[bl].find(blankIndex)->second;
+                            }
+                            newLogProb += logProbabilities[pos].find(targetD[targetIdx])->second;
+                            if (end == actualLogitLen) {
+                                for (int64_t ble = pos + 1; ble < actualLogitLen; ble++) {
+                                    newLogProb += logProbabilities[ble].find(blankIndex)->second;
+                                }
+                            }
+                            findPaths(nextIdx, pos + 1, end + 1, newLogProb);
+                        }
+                    } else {
+                        for (size_t pos = start; pos < end; pos++) {
+                            newLogProb = prevLogProb;
+                        size_t next_start = pos + 1;
+                        for (size_t bl = start; bl < pos; bl++) {
+                            newLogProb += logProbabilities[bl].find(blankIndex)->second;
+                        }
+                        if (end == actualLogitLen) {
+                            for (int64_t ble = pos + 1; ble < actualLogitLen; ble++) {
+                                newLogProb += logProbabilities[ble].find(blankIndex)->second;
+                            }
+                        }
+                        if (targetIdx < decodedTargetLen - 1
+                               && targetD[targetIdx] == targetD[targetIdx + 1]) {
+                            newLogProb += logProbabilities[next_start++].find(blankIndex)->second;
+                        }
+                        for (int64_t bl = pos; bl >= st64; bl--) {
+                            newLogProb += logProbabilities[bl].find(targetD[targetIdx])->second;
```

Comment:
what if `targetD[]` is not found then we have dereference error. I encountered this issue it may be due to incorrect input data generated in the OCR model by known bugs in ReduceSum. Nevertheless we should protect this operation from seg fault

Reply:
Fixed